### PR TITLE
dotnet 5.0

### PR DESCRIPTION
**Pull request by chenzhitong**

### CODE REVIEW

**Review by shargon — APPROVED** [review]
wait for preview4

File: Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  
+  <PropertyGroup>
+    <VersionPrefix>3.0.0</VersionPrefix>
+    <VersionSuffix>preview3</VersionSuffix>
```
Comment (vncoelho) [inline Directory.Build.props:6]: To change.

File: Dockerfile [hunk 2]
```diff
@@ -7,7 +7,7 @@ COPY NuGet.Config /neo-cli
 WORKDIR /neo-cli
 RUN dotnet restore && dotnet publish -c Release -o /app
 
-FROM mcr.microsoft.com/dotnet/core/runtime:3.0 AS Final
+FROM mcr.microsoft.com/dotnet/core/runtime:5.0.0 AS Final
 RUN apt-get update && apt-get install -y \
```
Comment (vncoelho) [inline Dockerfile:11]: `5.0.100` as well?

Comment (snowypowers) [reply Dockerfile:11]: Btw, they updated the namespaces for their docker images. See https://github.com/dotnet/dotnet-docker/issues/2375

The new one should be `mcr.microsoft.com/dotnet/runtime:5.0` (or whatever version you want)

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM mcr.microsoft.com/dotnet/core/sdk:3.0 AS Build
+FROM mcr.microsoft.com/dotnet/runtime:5.0.0 AS Build
```
Comment (snowypowers) [inline Dockerfile:1]:
```suggestion
FROM mcr.microsoft.com/dotnet/sdk:5.0.0 AS Build
```

File: Directory.Build.props (hunk 1 above)
Comment (shargon) [reply Directory.Build.props]: Done

**Review by shargon — DISMISSED** [review]

File: Directory.Build.props [hunk 4]
```diff
@@ -0,0 +1,16 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  
+  <PropertyGroup>
+    <VersionPrefix>3.0.0</VersionPrefix>
+    <VersionSuffix>preview4</VersionSuffix>
```
Comment (vncoelho) [inline Directory.Build.props:6]: change `VersionSuffix`?

File: neo-cli/neo-cli.csproj [hunk 5]
```diff
@@ -1,16 +1,11 @@
 <Project Sdk="Microsoft.NET.Sdk">
```
Comment (vncoelho) [inline neo-cli/neo-cli.csproj:1]: Change `<PackageReference Include="Neo" Version="3.0.0-preview4" />` ?

File: .github/workflows/dotnetcore.yml [hunk 6]
```diff
@@ -22,8 +22,8 @@ jobs:
     - name: Check format
       if: runner.os == 'Linux'
       run: |
-        dotnet tool install --version 3.2.111002 --tool-path ./ dotnet-format --add-source https://dotnet.myget.org/F/format/api/v3/index.json
-        ./dotnet-format --check --dry-run -v diagnostic
+        dotnet tool install --version 5.0.142902 --tool-path ./ dotnet-format --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json
+        ./dotnet-format neo-cli --check -v diagnostic
```
Comment (erikzhang) [inline .github/workflows/dotnetcore.yml:26]:
```suggestion
        ./dotnet-format --check -v diagnostic
```

File: Directory.Build.props [hunk 7]
```diff
@@ -0,0 +1,16 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (erikzhang) [inline Directory.Build.props:1]: I think we don't need `Directory.Build.props`.

**Review by erikzhang — CHANGES_REQUESTED** [review]

**Review by erikzhang — APPROVED** [review]

**Pull request merged**